contrib/parseq-retry/src/main/java/com/linkedin/parseq/retry/RetriableTask.java

```diff
@@ -0,0 +1,184 @@
+package com.linkedin.parseq.retry;
+
+import com.linkedin.parseq.BaseTask;
+import com.linkedin.parseq.Context;
+import com.linkedin.parseq.Priority;
+import com.linkedin.parseq.Task;
+import com.linkedin.parseq.function.Failure;
+import com.linkedin.parseq.function.Success;
+import com.linkedin.parseq.function.Try;
+import com.linkedin.parseq.internal.ArgumentUtil;
+import com.linkedin.parseq.promise.Promise;
+import com.linkedin.parseq.promise.Promises;
+import com.linkedin.parseq.promise.SettablePromise;
+
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+
+/**
+ * A parseq task wrapper that supports arbitrary retry policies.
+ *
+ * @author Oleg Anashkin (oleg.anashkin@gmail.com)
+ */
+public class RetriableTask<T> extends BaseTask<T> {
+  /** A name of the task that needs to be retried. */
+  protected final String _name;
+
+  /** A task generator function. It will receive a zero-based attempt number as a parameter. */
+  protected final Function<Integer, Task<T>> _taskFunction;
+
+  /** Retry policy which will control this task's behavior. */
+  protected final AbstractRetryPolicy<T> _policy;
+
+  /** Start time for the very first attempt. */
+  protected long _startedAt;
+
+  /**
+   * A parseq task wrapper that supports arbitrary retry policies.
+   *
+   * @param name A name of the task that needs to be retried.
+   * @param taskFunction A task generator function. It will receive a zero-based attempt number as a parameter.
+   * @param policy Retry policy that will control this task's behavior.
+   */
+  public RetriableTask(String name, Function<Integer, Task<T>> taskFunction, AbstractRetryPolicy<T> policy)
+  {
+    ArgumentUtil.requireNotNull(taskFunction, "taskFunction");
+    ArgumentUtil.requireNotNull(policy, "policy");
+
+    _name = name;
+    _taskFunction = taskFunction;
+    _policy = policy;
+  }
+
+  /**
+   * A parseq task wrapper that supports arbitrary retry policies.
+   *
+   * @param taskFunction A task generator function. It will receive a zero-based attempt number as a parameter.
+   * @param policy Retry policy that will control this task's behavior.
+   */
+  public RetriableTask(Function<Integer, Task<T>> taskFunction, AbstractRetryPolicy<T> policy)
+  {
+    this("operation", taskFunction, policy);
```

Comment:
This is the name of operation, not the task. It's used in logging to construct messages like "attempt 2 of <operationName> failed because of <reason>", so default "operation" name reads much better in this context than "withRetry".